conans/client/rest/rest_client.py

```diff
@@ -176,6 +177,11 @@ def upload_recipe(self, conan_reference, the_files, retry, retry_wait, ignore_de
 
         if not new and not deleted and modified == ["conanmanifest.txt"]:
             return False
+
+        if no_overwrite in ["all", "recipe"] and (modified not in (["conanmanifest.txt"], []) or
```

Comment:
I think in this line you know there are modifications, no need to check new, deleted and modified again?